### ISSUE TITLE
 Searchable transcript

### ISSUE THREAD

**Issue by jcoyne** [sul-dlss/sul-embed#1807]
Ideally, via a Searchbar in the transcript panel. Or, via Ctrl+F in the web browser. A match for the user's search query jumps the transcript to the corresponding segment, and highlights the matching string.

See the "Transcript Display" and "Single Companion Window Desktop" prototypes in the [Figma Design](https://www.figma.com/file/ADNitlk4aWHDiPDWD1dHdF/Media-Work-Cycle---Media-Player?type=design&node-id=0-1&mode=design
). [Click the "Present" play arrow in the top-right corner and select these prototypes.]
![Image](https://github.com/sul-dlss/sul-embed/assets/2343316/f8a89e8b-d3f7-4bcf-84fd-a39ac3d2cd02)

**Comment by jcoyne** [comment]
@alundgard How should this look if there are multiple matches?  What has precedence if it's on auto-scroll and a match is found at the same time, which item should scroll into view?

**Comment by alundgard** [comment]
Good question. Zoom handles it this way: the search query takes precedence over auto-scroll. A positive match automatically jumps the video playback to the (first) corresponding time segment, and auto-scroll proceeds from that segment. If there are multiple matches, each match is highlighted and the user can jump between matches using Up and Down arrow keys.
<img width="346" alt="zoom-transcript-search" src="https://github.com/sul-dlss/sul-embed/assets/2343316/425a3b69-76db-48a5-a80f-5f9e07180431">

**Comment by jcoyne** [comment]
Potentially the matches are many pages apart and they won't know that multiple matches exist.

**Comment by alundgard** [comment]
Yes, potentially. (These questions are also true of Zoom's transcript UI.) Positive results -> jump to match. No results -> nothing happens.
Also considered adding Up/Down arrows that become active upon positive match (similar to within-file search in VSCode), but space is so limited. Didn't want to clutter the UI too much, but happy to reconsider that.
<img width="317" alt="transcript-search-arrows" src="https://github.com/sul-dlss/sul-embed/assets/2343316/8eac1bd0-756c-42f3-8a92-d7eeb65d3899">

**Comment by alundgard** [comment]
Here's a quick mockup that surfaces more of the Search functionality in the UI. (In this case, No results -> shows 0/0 in the search field and the buttons are inactive.)
<img width="315" alt="transcript-search-result" src="https://github.com/sul-dlss/sul-embed/assets/2343316/12cd162b-589c-48d7-b92a-05886e0550b5">
Also worth noting: it looks like the browser's built-in search works for searching/navigating the transcript. Nice! Example: https://sul-purl-stage.stanford.edu/xs884bk3832.

**Comment by jcoyne** [comment]
If browser search works on this, is it a good idea to duplicate this? This will definitely create a maintenance burden, and will potentially make the UI more confusing. Is there a demonstrated user need?

**Comment by alundgard** [comment]
Maybe we should wait to see how the browser's built-in search interacts with the the auto-scroll functionality. But I think it's probably OK to deprioritize the within-player search (I ranked it as lowest-priority in the original transcript issue #1710). cc @dinahhandel @andrewjbtw
